crates/node/core/src/version.rs

```diff
@@ -1,63 +1,61 @@
 
-/// The human readable name of the client
-pub const NAME_CLIENT: &str = "Reth";
-
-/// The latest version from Cargo.toml.
-pub const CARGO_PKG_VERSION: &str = env!("CARGO_PKG_VERSION");
-
-/// The full SHA of the latest commit.
-pub const VERGEN_GIT_SHA_LONG: &str = env!("VERGEN_GIT_SHA");
-
-/// The 8 character short SHA of the latest commit.
-pub const VERGEN_GIT_SHA: &str = env!("VERGEN_GIT_SHA_SHORT");
-
-/// The build timestamp.
-pub const VERGEN_BUILD_TIMESTAMP: &str = env!("VERGEN_BUILD_TIMESTAMP");
-
-/// The target triple.
-pub const VERGEN_CARGO_TARGET_TRIPLE: &str = env!("VERGEN_CARGO_TARGET_TRIPLE");
-
-/// The build features.
-pub const VERGEN_CARGO_FEATURES: &str = env!("VERGEN_CARGO_FEATURES");
-
-/// The short version information for reth.
-pub const SHORT_VERSION: &str = env!("RETH_SHORT_VERSION");
-
-/// The long version information for reth.
-pub const LONG_VERSION: &str = concat!(
-    env!("RETH_LONG_VERSION_0"),
-    "\n",
-    env!("RETH_LONG_VERSION_1"),
-    "\n",
-    env!("RETH_LONG_VERSION_2"),
-    "\n",
-    env!("RETH_LONG_VERSION_3"),
-    "\n",
-    env!("RETH_LONG_VERSION_4")
-);
-
-/// The build profile name.
-pub const BUILD_PROFILE_NAME: &str = env!("RETH_BUILD_PROFILE");
-
-/// The version information for reth formatted for P2P (devp2p).
-///
-/// - The latest version from Cargo.toml
-/// - The target triple
-///
-/// # Example
-///
-/// ```text
-/// reth/v{major}.{minor}.{patch}-{sha1}/{target}
-/// ```
-/// e.g.: `reth/v0.1.0-alpha.1-428a6dc2f/aarch64-apple-darwin`
-pub(crate) const P2P_CLIENT_VERSION: &str = env!("RETH_P2P_CLIENT_VERSION");
+/// Constants for reth-cli
+#[derive(Debug, Default)]
+pub struct RethCliVersionConsts<'a> {
+    /// The human readable name of the client
+    pub name_client: &'a str,
+
+    /// The latest version from Cargo.toml.
+    pub cargo_pkg_version: &'a str,
+
+    /// The full SHA of the latest commit.
+    pub vergen_git_sha_long: &'a str,
+
+    /// The 8 character short SHA of the latest commit.
+    pub vergen_git_sha: &'a str,
+
+    /// The build timestamp.
+    pub vergen_build_timestamp: &'a str,
+
+    /// The target triple.
+    pub vergen_cargo_target_triple: &'a str,
+
+    /// The build features.
+    pub vergen_cargo_features: &'a str,
+
+    /// The short version information for reth.
+    pub short_version: &'a str,
+
+    /// The long version information for reth.
+    pub long_version: &'a str,
+    /// The build profile name.
+    pub build_profile_name: &'a str,
+
+    /// The version information for reth formatted for P2P (devp2p).
+    ///
+    /// - The latest version from Cargo.toml
+    /// - The target triple
+    ///
+    /// # Example
+    ///
+    /// ```text
+    /// reth/v{major}.{minor}.{patch}-{sha1}/{target}
+    /// ```
+    /// e.g.: `reth/v0.1.0-alpha.1-428a6dc2f/aarch64-apple-darwin`
+    pub p2p_client_version: &'a str,
+
+    /// extra data used for payload building
+    pub extra_data: String,
```

Comment:
I'd like to make all of these fields a `Cow<'static` then this can also be configured with `String`